library/alloc/src/vec.rs

```diff
@@ -1257,6 +1257,14 @@ impl<T> Vec<T> {
     #[inline]
     #[stable(feature = "append", since = "1.4.0")]
     pub fn append(&mut self, other: &mut Self) {
```

Comment:
Or we could just put a big notice that possible allocation stealing in the docs of `append` to give users a note.